template/alpha_formula.py

```diff
@@ -23,24 +24,29 @@ def __init__(self, _startdate, data, parameters):
         self.mid_last_k_days = {
             instrument: deque(
                 (datum.buy + datum.sell) / 2
-                for datum in price_data[-self.k_days - 1: -1])
+                if datum.buy is not None and datum.sell is not None else 0
+                for datum in price_data[-self.k_days - 1: -1]
+                )
             for instrument, price_data in data['currency_price_tw'].items()
         }
         self.sum_last_k_days = {
             instrument: sum(self.mid_last_k_days[instrument])
             for instrument in data['currency_price_tw']
         }
 
-    def generate(self, date, data):
+    def generate(self, _date, data):
         """
         Generate portfolio for the given date.
         The simulator guarantees to call this method in date order, and the
         first date will be "startdate" specified in AlphaSettings.
         """
-        portfolio = Portfolio(date)
-        for instrument, price_data in data['currency_price_tw'].items():
+        portfolio = Portfolio()
+        for instrument in TRADING_INSTRUMENTS[('currency', 'TW')]:
```

Comment:
The TRADING_INSTRUMENT may not be the same as instruments in price_data.
Hmm... finally think this judgment should be put in simulator and raise error. I'll fix it later.